property/index.js

```diff
@@ -112,53 +121,158 @@ module.exports = yeoman.Base.extend({
+          return this.askForParameters();
+      }
+
+      try {
+        debug('property definition input: %j', this.propDefinition);
+        coerceDefaultValue(this.propDefinition, answers.defaultValue);
+        debug('property coercion output: %j', this.propDefinition);
+      } catch (err) {
+        debug('Failed to coerce property default value: ', err);
+        this.log('Warning: please enter the ' + this.name +
+          ' property again. The default value provided "' +
+          answers.defaultValue + 
+          '" is not valid for the selected type: ' + this.type);
+        return this.askForParameters();
+      }
     }.bind(this));
   },
 
   property: function() {
     var done = this.async();
-    var def = {
-      name: this.name,
-      type: this.type
-    };
-    if (this.required) {
-      def.required = true;
-    }
-    if (this.defaultValue) {
-      if (this.type === 'boolean'){
-        if (['true', '1', 't'].indexOf(this.defaultValue) !== -1 ){
-          def.default = true;
-        } else {
-          def.default = false;
-        }
-      } else if (this.defaultValue === 'uuid' || this.defaultValue === 'guid'){
-         def.defaultFn = this.defaultValue;
-      } else if ((this.type === 'date' || this.type === 'datetime') &&
-                  this.defaultValue.toLowerCase() === 'now'){
-         def.defaultFn = 'now';
-      } else {
-         def.default = this.defaultValue;
-      }
-    }
-
-    this.modelDefinition.properties.create(def, function(err) {
+    this.modelDefinition.properties.create(this.propDefinition, function(err) {
       helpers.reportValidationError(err, this.log);
       return done(err);
     }.bind(this));
   },
-
   saveProject: actions.saveProject
 });
+
+function coerceDefaultValue(propDef, value) {
+  var itemType;
+  if (Array.isArray(propDef.type)) {
+    itemType = propDef.type[0];
+    propDef.type = 'array';
+  }
+
+  switch (propDef.type) {
+    case 'string':
+      if (value === 'uuid' || value === 'guid'){
+        propDef.defaultFn = value;
+      } else {
+        propDef.default = value;
+      }
+      break;
+    case 'number':
+      propDef.default = castToNumber(value);
+      break;
+    case 'boolean':
+      propDef.default = castToBoolean(value);
+      break;
+    case 'date':
+      if (value.toLowerCase() === 'now'){
+        propDef.defaultFn = 'now';
+      } else {
+        propDef.default = castToDate(value);
+      }
+      break;
+    case 'array':
+      propDef.type = [itemType];
+      if (itemType === 'string') {
+        propDef.default = value.replace(/[\s,]+/g, ',').split(',');
+      } else if (itemType === 'number') {
+        propDef.default = value.replace(/[\s,]+/g, ',').split(',')
+          .map(function(item) {
+            return castToNumber(item);
+          });
+      } else if (itemType === 'boolean') {
+        propDef.default = value.replace(/[\s,]+/g, ',').split(',')
+          .map(function(item) {
```

Comment:
I mean lines 223-225, to make myself more clear.

Instead of 

```js
.map(function(item) {
  return castToBoolean(item);
});
```

one ca write

```js
.map(castToBoolean);
```